### PR TITLE
WT-13858 Add early return from file open in live restore

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

**Review by ajmorton — APPROVED** [review]
LGTM

File: src/live_restore/live_restore_fs.c [hunk 1]
```diff
@@ -769,7 +769,8 @@ __live_restore_fh_find_holes_in_dest_file(
 
     data_end_offset = 0;
     WT_SYSCALL(((fd = open(filename, O_RDONLY)) == -1 ? -1 : 0), ret);
-    WT_ERR(ret);
+    if (ret != 0)
```
Comment (luke-pearson) [inline src/live_restore/live_restore_fs.c:772]: This should be WT_RET(ret);

Comment (luke-pearson) [reply src/live_restore/live_restore_fs.c:772]: Make that WT_RET_MSG(session, ret, "Failed to open file descriptor on %s", filename);

Comment (Sean04) [reply src/live_restore/live_restore_fs.c:772]: https://github.com/wiredtiger/wiredtiger/pull/11331/commits/2407de456246f88bd3871751eef396e3b5539f8a

Comment (Sean04) [reply src/live_restore/live_restore_fs.c:772]: Please correct me if I'm wrong, but I don't see `WT_RET_MSG` checking this return value in the same way `WT_RET` does. How can we remove this `if (ret != 0)`?

Comment (luke-pearson) [reply src/live_restore/live_restore_fs.c:772]: Oh what, I had assumed they do WT_RET+MSG. My b.

**Pull request merged**